compiler/rustc_typeck/src/check/check.rs

```diff
@@ -87,8 +87,26 @@ pub(super) fn check_fn<'a, 'tcx>(
 
     let declared_ret_ty = fn_sig.output();
 
-    let revealed_ret_ty =
-        fcx.instantiate_opaque_types_from_value(fn_id, declared_ret_ty, decl.output.span());
+    let feature = match tcx.hir().get(fn_id) {
+        Node::Item(hir::Item { kind: ItemKind::Fn(..), .. }) |
+        Node::ImplItem(hir::ImplItem {
+            kind: hir::ImplItemKind::Fn(..), ..
+        }) => None,
+        // I don't know if TAIT uses in trait declarations make sense at all
```

Comment:
I'd probably just drop this comment -- save it for the RFC thread, @oli-obk! =)

PS this is my silly way of telling you I think they do ;)